### ISSUE TITLE
Refactor of `Algebra/Order/LatticeGroup` with `Algebra/Order/Group/Abs`

### ISSUE THREAD

**Issue by mans0954** [leanprover-community/mathlib4#6376]
The additive forms of several results in `Algebra/Order/LatticeGroup` have corresponding results in `Algebra/Order/Group/Abs`. E.g. compare the additive forms of the following results from `Algebra/Order/LatticeGroup`:
```
theorem le_mabs (a : α) : a ≤ |a| := sorry
theorem inv_le_abs (a : α) : a⁻¹ ≤ |a| := sorry
theorem one_le_abs [CovariantClass α α (· * ·) (· ≤ ·)] (a : α) : 1 ≤ |a| := sorry
theorem mabs_mul_le [CovariantClass α α (· * ·) (· ≤ ·)] (a b : α) : |a * b| ≤ |a| * |b| := sorry
theorem mabs_mabs [CovariantClass α α (· * ·) (· ≤ ·)] (a : α) : |(|a|)| = |a| := sorry
theorem mabs_of_one_le [CovariantClass α α (· * ·) (· ≤ ·)] (a : α) (h : 1 ≤ a) : |a| = a := sorry
theorem abs_div_comm (a b : α) : |a / b| = |b / a| := sorry
theorem abs_abs_div_abs_le [CovariantClass α α (· * ·) (· ≤ ·)] (a b : α) : |(|a| / |b|)| ≤ |a / b| := sorry
theorem abs_inv (a : α) : |a⁻¹| = |a| := sorry
```
with these results from `Algebra/Order/Group/Abs`:
```
theorem le_abs_self (a : α) : a ≤ |a| := sorry
theorem neg_le_abs_self (a : α) : -a ≤ |a| := sorry
theorem abs_nonneg (a : α) : 0 ≤ |a| := sorry
theorem abs_add (a b : α) : |a + b| ≤ |a| + |b| := sorry
theorem abs_abs (a : α) : |(|a|)| = |a| := sorry
theorem abs_of_nonneg (h : 0 ≤ a) : |a| = a := sorry
theorem abs_sub_comm (a b : α) : |a - b| = |b - a| := sorry
theorem abs_abs_sub_abs_le_abs_sub (a b : α) : |(|a| - |b|)| ≤ |a - b| := sorry
theorem abs_neg (a : α) : |(-a)| = |a| := sorry
```

Would it make sense to refactor this?

**Comment by mans0954** [comment]
Probably the thing to do would be to move some of these results earlier in `Abs` and weaken `LinearOrder` to `PartialOrder`:
e.g.

```
section Div

-- see Note [lower instance priority]
/-- `abs a` is the absolute value of `a`. -/
@[to_additive "`abs a` is the absolute value of `a`"]
instance (priority := 100) Inv.toHasAbs [Inv α] [Sup α] : Abs α :=
  ⟨fun a => a ⊔ a⁻¹⟩
#align has_inv.to_has_abs Inv.toHasAbs
#align has_neg.to_has_abs Neg.toHasAbs

@[to_additive]
theorem abs_eq_sup_inv [Inv α] [Sup α] (a : α) : |a| = a ⊔ a⁻¹ :=
  rfl
#align abs_eq_sup_inv abs_eq_sup_inv
#align abs_eq_sup_neg abs_eq_sup_neg

variable [Inv α] [Lattice α] {a b : α}

@[to_additive le_abs_self]
theorem le_mabs_self (a : α) : a ≤ |a| :=
  le_sup_left
#align le_abs_self le_abs_self

@[to_additive]
theorem inv_le_abs_self (a : α) : a⁻¹  ≤ |a| :=
  le_sup_right
#align neg_le_abs_self neg_le_abs_self

end Div
```

**Comment by mans0954** [comment]
Actually, now I think about it, this requires developing some results from non-commutative lattice ordered groups. I may have this on a branch somewhere, let me look.

**Comment by mans0954** [comment]
@urkud [asks](https://github.com/leanprover-community/mathlib4/pull/6452#discussion_r1289154894) if we can do any of this for Monoids.

Which leads me to wonder how much requires associativity?

**Comment by YaelDillies** [comment]
I have an incoming PR which solves the titular issue. I'm not sure how to solve Yury's comment yet. Regarding associativity, I agree we probably don't need it much, but I also doubt we care about non-associative ordered things?

**Comment by YaelDillies** [comment]
Closed by #9553. If you want to make non-commutativity and non-associativity issues, please open a new issue (personally I think we don't care too much about them, so I'd happy to leave things as they are).
